src/installer.js

```diff
@@ -62,59 +61,54 @@ class RedhatInstaller extends common.ElectronInstaller {
     const dest = path.join(process.env.HOME, '.rpmmacros')
     this.options.logger(`Creating macros file at ${dest}`)
 
-    return common.createTemplatedFile(src, dest, Object.assign({ dir: this.stagingDir }, this.options))
-      .catch(common.wrapError('creating macros file'))
+    return common.wrapError('creating macros file', async () => common.createTemplatedFile(src, dest, Object.assign({ dir: this.stagingDir }, this.options)))
   }
 
   /**
    * Package everything using `rpmbuild`.
    */
-  createPackage () {
+  async createPackage () {
     this.options.logger(`Creating package at ${this.stagingDir}`)
 
-    return spawn('rpmbuild', ['-bb', this.specPath, '--target', this.options.arch], this.options.logger)
-      .then(output => this.options.logger(`rpmbuild output: ${output}`))
+    const output = await spawn('rpmbuild', ['-bb', this.specPath, '--target', this.options.arch], this.options.logger)
+    this.options.logger(`rpmbuild output: ${output}`)
   }
 
   /**
    * Creates the spec file for the package.
    *
    * See: https://fedoraproject.org/wiki/How_to_create_an_RPM_package
    */
-  createSpec () {
+  async createSpec () {
     const src = path.resolve(__dirname, '../resources/spec.ejs')
     this.options.logger(`Creating spec file at ${this.specPath}`)
 
-    return this.createTemplatedFile(src, this.specPath)
-      .catch(common.wrapError('creating spec file'))
+    return common.wrapError('creating spec file', async () => this.createTemplatedFile(src, this.specPath))
   }
 
   /**
    * Get the hash of default options for the installer. Some come from the info
    * read from `package.json`, and some are hardcoded.
    */
-  generateDefaults () {
-    return common.readElectronVersion(this.userSupplied.src)
-      .then(electronVersion => Promise.all([
-        common.readMetadata(this.userSupplied),
-        redhatDependencies.forElectron(electronVersion, this.userSupplied.logger)
-      ])).then(([pkg, requires]) => {
-        pkg = pkg || {}
-
-        this.defaults = Object.assign(common.getDefaultsFromPackageJSON(pkg), {
-          version: pkg.version || '0.0.0',
-          license: pkg.license,
-          compressionLevel: 2,
-          icon: path.resolve(__dirname, '../resources/icon.png'),
-
-          pre: undefined,
-          post: undefined,
-          preun: undefined,
-          postun: undefined
-        }, requires)
-
-        return this.defaults
-      })
+  async generateDefaults () {
+    const electronVersion = await common.readElectronVersion(this.userSupplied.src)
+    const [pkg, requires] = await Promise.all([
+      (async () => (await common.readMetadata(this.userSupplied)) || {})(),
+      redhatDependencies.forElectron(electronVersion, this.userSupplied.logger)
+    ])
+    this.defaults = Object.assign(common.getDefaultsFromPackageJSON(pkg), {
```

Comment:
When using the master branch of `-common`, this line should account for the new revision behaivor
```suggestion
    this.defaults = Object.assign(common.getDefaultsFromPackageJSON(pkg, { revision: 1 }), {
```